### ISSUE TITLE
AzureCP configuration | File not found Error

### ISSUE THREAD

**Issue by JeganBas** [Yvand/AzureCP#102]
Hi,
To fix the people picker issue in my SharePoint site,  I've been instructed to download and install Azure CP  component in my single server SharePoint 2016 Farm ( on-premises ), after deploying Azure CP (v 17),  when i tried to test the connection by providing the client ID and Secret, I'm getting **File not found error,** which is very weird  ,because uploading certificate file (.pfx) is optional.

![E2](https://user-images.githubusercontent.com/45036764/95277012-5f199600-086a-11eb-9f61-9e5c02562b5e.jpg)
![E1](https://user-images.githubusercontent.com/45036764/95277016-617bf000-086a-11eb-90fa-3a7af198acca.jpg)
 
Above are the screenshot of the error.

Also, is there any way that i can download the AzureCP.wsp for v13 ?, if yes, can you please provide the download link . Else anyone please share the v13 file here.  Since that is the preferred version to be used as per the Job aid document shared with me in my project. 

Kindly help me to fix this issue.  Thanks

**Comment by Yvand** [comment]
Do you get this error when you try to test the connection, or add it?
Can you try to add it without testing it?
I confirm that the certificate is optional, so you don't need to specify one if you specify a client secret.
v17 has significant improvements over v13 so you should really keep it.
But if you really decide to roll back to a lower version, you can find the history in the [releases](https://github.com/Yvand/AzureCP/releases).

**Comment by JeganBas** [comment]
Hi @Yvand  thanks for replying,  I got the error while testing only.   When i proceed without testing I'm not getting the actual results, when i use the group email IDs in the SharePoint site.  It fails to resolve them.  Let me tryout with the low version and update you.

**Comment by Yvand** [comment]
if it doesn't work you can check the logs as instructed in https://yvand.github.io/AzureCP/Troubleshoot-AzureCP.html
Again, I don't recommend that you downgrade to v13

**Comment by JeganBas** [comment]
Hi @Yvand thanks for Sharing the link,  and i understand your concern, but still as per our team's recommendation we are trying to downgrade and testing the feature. 

This time we are unable to deploy the AzureCP itself.  when  i tried to deploy the WSP file as a farm solution, its taking more time than usual.  For more than 3 hours the status hasn’t changed from deploying. I even tried to remove the solution and added again, but still it results the same. Also I verified with other normal WSP files, they are all working fine and get deployed within minutes.

So request you to reply, what is the expected deployment time for AzureCP and does the above mentioned behavior is the expected one or not ?.

![image](https://user-images.githubusercontent.com/45036764/95447911-21a72c80-0980-11eb-9fdf-dcf89f999748.png)

**Comment by JeganBas** [comment]
Hi @Yvand , I've tried the given Fix setup issues procedures, but still my issue is not getting resolved.  The solution is keep on showing as deploying, it is not moving to deployed status. I tried multiple other ways, nothing helps.  

I even tried to install the v17 again, but even with v17 the status is same. never going to deployed status. 

When i stopped and restarted the deployment using manual options, i got the following error. 
![image](https://user-images.githubusercontent.com/45036764/95684553-22ee8880-0c10-11eb-96bc-c883febc11ac.png)

Please help us with the solution to fix this deployment issue, after this deployment only we need to work on our main issue.

**Comment by Yvand** [comment]
@JeganBas this error is exactly the scenario described in https://yvand.github.io/AzureCP/Fix-setup-issues.html
You can follow the step by step in this article to fix the issue properly and totally clean your farm

**Comment by JeganBas** [comment]
@Yvand , finally I've found the root cause of this issue, that is the .NET Framework version.
**When i bumped up my .NET Framework to 4.8 version, I was able to install and configure AzureCP smoothly**. so i think @Yvand need to update the pre-requisites for 
![cp](https://user-images.githubusercontent.com/45036764/96529478-e4f20400-12a2-11eb-82f3-3be1cc31b00f.jpg)
AzureCP to 4.8. I'd just thought of bringing this to your attention.
